src/GLuaFixer/LintMessage.hs

```diff
@@ -1,18 +1,60 @@
+{-# LANGUAGE OverloadedStrings #-}
 module GLuaFixer.LintMessage where
 
 import GLua.AG.Token
 import GLua.AG.PrettyPrint
+import Data.Aeson
+import Control.Monad
+import Data.Char
 import Data.List (sortOn)
+import Text.ParserCombinators.UU.BasicInstances hiding (msgs)
+
+data LogFormat = StandardLogFormat | GithubLogFormat
+
+instance Show LogFormat where
+  show (StandardLogFormat) = "standard"
+  show (GithubLogFormat) = "github"
+
+instance ToJSON LogFormat where
+  toJSON StandardLogFormat = "standard"
+  toJSON GithubLogFormat = "github"
+
+instance FromJSON LogFormat where
+  parseJSON (String logFormat) = case logFormat of
+    "standard" -> return ( StandardLogFormat )
+    "github" -> return ( GithubLogFormat )
+    _ -> fail ( "Please use either \"standard\" or \"github\" but was " ++ show logFormat )
+  parseJSON _ = mzero
 
 -- | Represents lint messages
 data LintMessage
   = LintError { err_rg :: Region, err_msg :: String, err_file :: String }
```

Comment:
This is actually an early mistake of mine. The two constructors duplicate the message, region and file. It would have been better to have done the following:

```haskell
data Severity = Warning | Error

data LintMessage = LintMessage {lintmsg_severity :: Severity, lintmsg_msg :: String, lintmsg_file :: String}
```

That would make the formatting code prettier as well. But alas, back in the day this was one of my first Haskell projects, so I had much to learn myself 😉